contributor-book/src/guides/adding-a-new-operation-to-burn.md

```diff
@@ -0,0 +1,216 @@
+2. for the
+   [numeric trait](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L1618)
+3. for the implementation of numeric for
+   [float](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L2186)
+   and
+   [int](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/numeric.rs#L1903)
+
+Tensor is a struct that has a single member: `primitive` (defined
+[here](https://github.com/tracel-ai/burn/blob/main/burn-tensor/src/tensor/api/base.rs)), that is
+defined by it's
+[`Kind`](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/api/kind.rs#L15):
+one of `Bool`, `Float`, or `Int` (those linked in 3). These call the ops for that data type defined
+in the
+[`Backend`](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/backend/base.rs#L52)
+supertrait[^1]. This is the trait that is then implemented by the different `burn-` backends (such
+as `burn-ndarray` and `burn-wgpu`) which implement the functions if no default is provided.
+
+In this case, we only need to worry about `Bool` Tensors. Ops for `Float` is implemented under
+[burn-tensor/src/tensor/ops/tensor.rs](https://github.com/tracel-ai/burn/blob/3b7d9feede702cd616c273fa9eba9fbf14f66964/burn-tensor/src/tensor/ops/tensor.rs#L873),
+and for `Int` under
+[`burn-tensor/src/tensor/ops/int_tensor.rs`](https://github.com/tracel-ai/burn/blob/e1d873abe2c2fa0bb316719c4761eaf796291166/burn-tensor/src/tensor/ops/int_tensor.rs#L486).
+The current convention is ops of each type, if not unique to that type, are prefixed with the type. So `powf` and sundry would be defined as `int_powf` for `IntTensorOps` and `float_powf` for `FloatTensorOps`. If an op is unique to a type, then it should be implemented under `burn-tensor/src/api/{type}.rs`. For example, here is an implementation for [`sin` under `burn-tensor/src/api/float.rs`](https://github.com/tracel-ai/burn/blob/2acf6561dc9e173870d4209ed40ebbdcf7e3888c/burn-tensor/src/tensor/api/float.rs#L78) which obviously doesn't make sense for `Int`  or `Bool` tensors.
+
+The `Int` Tensor function uses the ones defined for Float with 2 extra casts (LHS to a `Float` tensor,
+Output to an `Int`). Given that the rest of the code will only look at the float implementations.
+
+### Adding Test
+
+Additional Test should be added to `burn-tensor` under
+[`burn-tensor/src/tests/ops/{op_name}.rs`](https://github.com/tracel-ai/burn/burn-tensor/src/tests/ops/powf.rs),
+inserting the module name into `burn-tensor/src/tests/ops/mod.rs`. Then add it to the `testgen_all`
+macro under `burn-tensor/src/tests/mod.rs`. This test is automatically added to the backends so it
+isn't necessary to add them there, save for those that require specific testing such
```

Comment:
save => same

Reply:
I actually meant to say "you won't need to implement tests on the backends directly, save for those that require their own test not covered by `burn-tensor`", let me make this section a bit more clear